internal/hcs/watcher.go

```diff
@@ -17,14 +18,16 @@ import (
 // Usage is:
 //
 // completed := false
-// go syscallWatcher("some description", &completed)
+// go syscallWatcher(context, &completed)
 // <syscall>
 // completed = true
 //
-func syscallWatcher(description string, syscallCompleted *bool) {
+func syscallWatcher(context logrus.Fields, syscallCompleted *bool) {
 	time.Sleep(timeout.SyscallWatcher)
 	if *syscallCompleted {
 		return
 	}
-	logrus.Warnf("%s: Did not complete within %s. This may indicate a platform issue. If it appears to be making no forward progress, obtain the stacks and see is there is a syscall stuck in the platform API for a significant length of time.", description, timeout.SyscallWatcher)
+	logrus.WithFields(context).
+		WithField(logfields.Timeout, timeout.SyscallWatcher.Nanoseconds()).
```

Comment:
What happens if you just pass `timeout.SyscallWatcher`, without the `.Nanoseconds()`?

Regardless, having the log specify nanoseconds as an integer seems like it will be hard to read.

Reply:
Wouldn't we want the raw data to be in NS so that any viewer can display it in "pretty" print? But we don't want to drop the rounding early correct?

Reply:
Will very what the default for passing a `timeout` is though.

Reply:
Well, you're throwing the type information away here, so e.g. if we build TraceLogging support for logrus, then we can't specifically log this as a duration.

So I guess the question is what does the JSON logger do with `time.Time`?